online/src/main/scala/ai/chronon/online/FetcherBase.scala

```diff
@@ -286,29 +288,48 @@ class FetcherBase(kvStore: KVStore,
-              try {
-                if (debug)
-                  logger.info(
-                    s"Constructing response for groupBy: ${groupByServingInfo.groupByOps.metaData.getName} " +
-                      s"for keys: ${request.keys}")
-                constructGroupByResponse(batchResponseTryAll,
-                                         streamingResponsesOpt,
-                                         groupByServingInfo,
-                                         queryTs,
-                                         startTimeMs,
-                                         multiGetMillis,
-                                         context,
-                                         totalResponseValueBytes)
-              } catch {
-                case ex: Exception =>
-                  // not all exceptions are due to stale schema, so we want to control how often we hit kv store
-                  getGroupByServingInfo.refresh(groupByServingInfo.groupByOps.metaData.name)
-                  context.incrementException(ex)
-                  ex.printStackTrace()
-                  throw ex
+              val groupByResponse: Map[String, AnyRef] =
+                try {
+                  if (debug)
+                    logger.info(
+                      s"Constructing response for groupBy: ${groupByServingInfo.groupByOps.metaData.getName} " +
+                        s"for keys: ${request.keys}")
+                  constructGroupByResponse(batchResponseTryAll,
+                                           streamingResponsesOpt,
+                                           groupByServingInfo,
+                                           queryTs,
+                                           startTimeMs,
+                                           multiGetMillis,
+                                           context,
+                                           totalResponseValueBytes)
+                } catch {
+                  case ex: Exception =>
+                    // not all exceptions are due to stale schema, so we want to control how often we hit kv store
+                    getGroupByServingInfo.refresh(groupByServingInfo.groupByOps.metaData.name)
+                    context.incrementException(ex)
+                    ex.printStackTrace()
+                    throw ex
+                }
+              if (groupByServingInfo.groupBy.hasDerivations) {
+                val derivedMapTry: Try[Map[String, AnyRef]] = Try {
+                  val deriveFunc: DerivationFunc = getGroupByServingInfo(request.name).get.deriveFunc
+                  applyDeriveFunc(deriveFunc, request, groupByResponse)
+                }
+                val derivedMap = derivedMapTry match {
+                  case Success(derivedMap) =>
+                    derivedMap
+                  // If the derivation failed we want to return the exception map only
+                  case Failure(exception) => {
+                    val derivedExceptionMap =
```

Comment:
optional: do we want to also use the renaming logic here?

Reply:
updated